### PR TITLE
Fix BOM visibility conditions and start using vitest for frontend unit tests

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
#429 made projects with purchasable parts don't render BOM.
This PR fixes this regression and tests the visibility of BOM for projects with and without purchasable parts.

### CODE REVIEW

**Comment by kasbah** [comment]
This is the kind of thing where I don't want to keep adding e2e tests. Feels like this should be a unit test on the component.

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
         "reservedFirst": true
       }
     ],
-    "react/jsx-filename-extension": [1, { "extensions": [".js"] }],
+    "react/jsx-filename-extension": [1, { "extensions": [".jsx", ".tsx"] }],
```
Comment (kasbah) [inline frontend/.eslintrc.json:52]: Is the .jsx rename necessary?

**Review by kasbah — CHANGES_REQUESTED** [review]
Looks good, noticed some minor things...

File: frontend/.eslintrc.json (hunk 1 above)
Comment (AbdulrhmnGhanem) [reply frontend/.eslintrc.json:52]: Using JSX in `.js` files complicated vitest setup. See https://github.com/vitejs/vite/discussions/3448

File: frontend/.nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+v16
```
[the remark this answers is not in the document]
Comment (kasbah) [reply frontend/.nvmrc:1]: But fine, if it's useful let's add it.

**Comment by kasbah** [comment]
Can you give this a better title please?

File: frontend/__tests__/BuyParts.test.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { afterEach, beforeEach, expect, it } from 'vitest'
```
Comment (kasbah) [inline frontend/__tests__/BuyParts.test.tsx:1]: Can this folder just be named `tests`? `__tests__` seems unnecessarily annoying.

Comment (AbdulrhmnGhanem) [reply frontend/__tests__/BuyParts.test.tsx]: Resolved in [039b270](https://github.com/kitspace/kitspace-v2/pull/446/commits/039b270cbdfd8d6c0b5f0b270beed0bd23e54fee).

**Comment by kasbah** [comment]
.jsx needs to be included in tsconfig

File: frontend/__tests__/BuyParts.test.tsx (hunk 3 above)
Comment (kasbah) [reply frontend/__tests__/BuyParts.test.tsx]: Nice, but sorry, should have said `test` to match the processor.

**Comment by AbdulrhmnGhanem** [comment]
> .jsx needs to be included in tsconfig

I added it in #466.

**Comment by kasbah** [comment]
Can you cherry-pick it here or something? Because the rename of the files is happening here. So this is PR as it stands is breaking typechecking if I understand it correctly.

File: frontend/tsconfig.json [hunk 4]
```diff
@@ -23,6 +23,6 @@
     "isolatedModules": true,
     "jsx": "preserve"
   },
-  "include": ["**/*.js", "**/*.ts", "**/*.tsx", "next-env.d.ts"],
+  "include": ["**/*.js", "**/*.ts", "**/*.tsx", "**/*.jsx", "next-env.d.ts"],
```
Comment (kasbah) [inline frontend/tsconfig.json:26]:
```suggestion
  "include": ["**/*.js", "**/*.jsx", "**/*.ts", "**/*.tsx", "next-env.d.ts"],
```

**Pull request merged**